app/src/main/kotlin/com/adesso/movee/scene/login/LoginViewModel.kt

```diff
@@ -28,19 +28,13 @@ class LoginViewModel @Inject constructor(
     }
 
     fun onLoginClick() {
-        uiScope.launch {
-            val username = username.value ?: return@launch
-            val password = password.value ?: return@launch
-
-            _loginInProgress.value = true
-
-            val loginResult = onBackgroundThread {
-                loginUseCase.run(LoginUseCase.Params(username, password))
-            }
-
-            loginResult.either(::handleFailure, ::navigateHome)
+        _loginInProgress.value = true
 
+        val username = username.value ?: return
+        val password = password.value ?: return
```

Comment:
line 31 should be below line 33 & 34 since one of them could return from the function and _loginInProgress.value would hang with true